core/base/inc/TStringLong.h

```diff
@@ -50,6 +52,8 @@ class TStringLong : public TString {
    virtual Int_t    Sizeof() const;
 
    ClassDef(TStringLong,1)  //Long string class (more than 255 chars)
-};
+} R__SUGGEST_ALTERNATIVE("TString");
```

Comment:
Shouldn't this be `R__ALWAYS_SUGGEST_ALTERNATIVE` and we get rid of the `_LOCAL` part?

Reply:
but how would you avoid having deprecation warnings for all deprecated classes?

Reply:
I probably misunderstand what this is supposed to do.

Right now, whatever `#include`s `ROOT/RConfig.hxx` first, gets do define what `R__SUGGEST_NEW_INTERFACE_LOCAL` is and which effect it has.

`TStringLong.h` defines it around the class definition, unconditionally - whatever users do.

I guessed that the intent is that `TSringLong` always warns, on any use. In that case we could remove `#define ` / `#undef R__SUGGEST_NEW_INTERFACE_LOCAL`, and simply use a `R__ALWAYS_SUGGEST_ALTERNATIVE` that always complains, no matter what's the value of `R__SUGGEST_NEW_INTERFACE`.

Reply:
That might actually work. You would go into hell, though, when you flag an often-used class with this.

Reply:
@Axel-Naumann now I see, it makes sense. One of the intent of this PR was really to see the deprecation in action from another perspective to see how to improve, and it's working!
Changes coming (not force-pushed ;-) )